backend/app/analysis/text_to_music.py

```diff
@@ -0,0 +1,179 @@
+import string
+from nltk.sentiment.vader import SentimentIntensityAnalyzer
+import numpy as np
+from scipy.io import wavfile
+import io
+import base64
+from nltk.tokenize import sent_tokenize
+import random
+
+
+musical_chars = {'a', 'b', 'c', 'd', 'e', 'f', 'g'}
+mc_list = list(musical_chars)
+
+# frequencies found from https://pages.mtu.edu/~suits/notefreqs.html
+note_freqs = {
+    'a': 440,
+    'b': 494,
+    'c': 523,
+    'd': 587,
+    'e': 330,
+    'f': 349,
+    'g': 392
+}
+
+# Ratios found from Wikipedia
+# (https://en.wikipedia.org/wiki/Consonance_and_dissonance#/media/File:Dyadic_harmonic_entropy_graph_(optimized_for_low_resolution).png)
+dissonant_ratios = [(5, 6), (4, 7), (5, 8), (5, 7), (6, 7)]
+neutral_ratios = [(3, 4), (3, 5), (4, 5)]
+consonant_ratios = [(1, 2), (2, 3)]
+
+
+def analyse_sentiment(text):
+    """
+    :param text: a string of text
+    :return: analysis of the text
+    """
+    # Convert text to lowercase
+    lower_case = text.lower()
+    # Removing punctuations
+    cleaned_text = lower_case.translate(str.maketrans('', '', string.punctuation))
+    score = SentimentIntensityAnalyzer().polarity_scores(cleaned_text)
+    return score
+
+
+def text_to_note(text):
+    """
+    :param text: Takes in a string of text
+    :return: A sound based on how negative or positive the text is.
+    """
+    score = analyse_sentiment(text)
+    note_freq = 400 + (score["pos"] - score["neg"]) * 350 * (1 + score["neu"])
+
+    # get timesteps for the sample, T is note duration in seconds
+    sample_rate = 44100
+    T = 1
+    t = np.linspace(0, T, T * sample_rate, False)
+
+    # generate sine wave notes
+    note = np.sin(note_freq * t * 2 * np.pi)
+
+    # concatenate notes
+    audio = note
+    # normalize to 16-bit range
+    audio *= 32767 / np.max(np.abs(audio))
+    # convert to 16-bit data
+    audio = audio.astype(np.int16)
+
+    return wav_to_base64(audio, sample_rate)
+
+
+def get_other_freq(score, current_freq):
+    """
+    :param score: the sentiment analysis score of a sentence
+    :param current_freq: the frequency of a note
+    :return other_freq: frequency for another note found from current_freq and a randomized ratio
+    """
+    if score['neu'] > score['pos'] and score['neu'] > score['neg']:
+        ratios = neutral_ratios
+    elif score['pos'] > score['neg']:
+        ratios = consonant_ratios
+    else:
+        ratios = dissonant_ratios
+
+    ratio = random.choices(ratios)[0]
+    other_freq = current_freq * ratio[0] / (ratio[1])
+    return other_freq
```

Comment:
I'd recommend not passing in the data structure `score`, but rather passing in the values of its properties (`neu`, `pos`, `neg`), which are simply percentage floats. Taken in isolation, this function requires that its input argument `score` have certain properties, or it'll not perform its function. The function as written, however, has no way of enforcing that those properties exist on that argument. These lines of code are straightforward and small, but in later work and as this code expands, it may become unintuitive what properties exactly are required for the code to function as expected. It's always best, if possible, to either introduce a typing system that would allow us to feel confident that a set of properties will always exist on our non-primitive arguments (say, by creating a `SentimentAnalysisScore` class in this file and using it to generate your `score` dictionary with validated properties), or to pass in primitive type arguments (strings, floats, integers, booleans, etc.). So that'd be something like:

```
def get_other_freq(positive_score, negative_score, neutral_score, current_frequency):
  etc.
```